backend/src/services/enhanced_database_service.py

```
@@ -226,6 +226,26 @@ async def save_trip_data(self, trip_data: TripData) -> StorageResult:
+            trip_data = await self.get_trip_data(trip_id)
+            if not trip_data:
+                logger.warning(f"Trip not found for saving guide: {trip_id}")
+                return False
+            
+            # Update with enhanced guide
+            trip_data.enhanced_guide = guide
+            trip_data.updated_at = datetime.now().isoformat()
+            
+            # Save updated trip data
+            result = await self.save_trip_data(trip_data)
+            return result.success if result else False
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)